### PR TITLE
Support different kind of args in min/max and update tests

### PR DESCRIPTION
**Pull request by HarshitaKalani**
Fixes: #4581 
Towards: #4017, #3067

### CODE REVIEW

**Comment by certik** [comment]
If this is an extension, then we need to give at least a warning. Eventually we need to control such extensions via compiler flags, but a warning for now should be enough.

**Comment by HarshitaKalani** [comment]
Alright. I'll do that. Thanks

File: tests/reference/asr-intrinsics13-349f6ac.stderr [hunk 1]
```diff
@@ -0,0 +1,5 @@
+semantic error: All arguments to min0 must be of the same type
+ --> tests/errors/intrinsics13.f90:2:14
+  |
+2 |     print *, min(12, 13.94)
+  |              ^^^^^^^^^^^^^^
```
Comment (certik) [inline tests/reference/asr-intrinsics13-349f6ac.stderr:5]: Is this file still current? I thought this PR makes this compile, with a warning.

File: tests/reference/asr-max_01-b3ef694.stderr [hunk 2]
```diff
@@ -1,5 +1,5 @@
-semantic error: All arguments to max0 must be of the same type and kind
+warning: Different kinds of args in max0 is a non-standard extension
  --> tests/errors/max_01.f90:6:14
   |
 6 |     print *, max(y, z)
-  |              ^^^^^^^^^ 
+  |              ^^^^^^^^^ Make kind of the all the arguments same
```
Comment (certik) [inline tests/reference/asr-max_01-b3ef694.stderr:5]: Maybe something like this:
```suggestion
  |              ^^^^^^^^^ help: ensure all arguments have the same kind to make it standard
```

There is also a typo `the all the`.

**Review by certik — APPROVED** [review]
Just some minor comments, overall good. Thanks!

File: tests/reference/asr-intrinsics13-349f6ac.stderr (hunk 1 above)
Comment (HarshitaKalani) [reply tests/reference/asr-intrinsics13-349f6ac.stderr:5]: No actually. This PR will allow args of same type but different kind to compile. Here the types of arg is different (int and real) which is not allowed.

Comment (certik) [reply tests/reference/asr-intrinsics13-349f6ac.stderr:5]: Ah I see, perfect.

**Pull request merged**